crux-http-server/src/crux/http_server/entity.clj

```diff
@@ -60,129 +50,55 @@
-                                                      :title "/entity"
-                                                      :options opts})
-                           not-found? (let [not-found-message (str eid " entity not found")]
-                                        (util/raw-html {:title "/entity"
-                                                        :body [:div.error-box not-found-message]
-                                                        :options opts
-                                                        :results {:entity-results
-                                                                  {"error" not-found-message}}}))
-                           error (let [error-message (.getMessage ^Exception error)]
-                                   (util/raw-html {:title "/entity"
-                                                   :body [:div.error-box error-message]
-                                                   :options opts
-                                                   :results {:entity-results
-                                                             {"error" error-message}}}))
-                           entity-history (try
-                                            (util/raw-html {:body (entity-history->html res)
-                                                            :title "/entity?history=true"
-                                                            :options opts
-                                                            :results {:entity-results (iterator-seq entity-history)}})
-                                            (finally
-                                              (.close entity-history)))
-                           :else (util/raw-html {:body (entity->html res)
-                                                 :title "/entity"
-                                                 :options opts
-                                                 :results {:entity-results entity}}))]
+    (encode-to-bytes [_ {:keys [eid no-entity? not-found? cause entity ^Closeable entity-history] :as res} charset]
+      (let [^String resp (util/raw-html {:title "/_crux/entity"
+                                         :options opts
+                                         :results (cond
+                                                    no-entity? nil
+                                                    not-found? {:entity-results {"error" (str eid " entity not found")}}
+                                                    cause {:entity-results {"error" cause}}
+                                                    entity-history (try
+                                                                     {:entity-results (iterator-seq entity-history)}
+                                                                     (finally
+                                                                       (.close entity-history)))
```

Comment:
you can't close `entity-history` before the call to `raw-html` has returned, because `entity-history` is lazy - you'd be closing it before it's consumed

Reply:
Ahhhh - think theres a similar problem within query, there, too! WIll fix :+1: